pkg/util/error_test.go

```diff
@@ -57,3 +57,53 @@ func TestCheckErr(t *testing.T) {
 		})
 	}
 }
+
+func TestHandleEmptyTableError(t *testing.T) {
+	type args struct {
+		resource string
+		ns       string
+		casType  string
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			"No Namespace and cas",
+			args{
+				resource: "ResourceType",
+				ns:       "",
+				casType:  "",
+			},
+		},
+		{
+			"Wrong cas or Namespace",
+			args{
+				resource: "ResourceType",
+				ns:       "InValidNamespace",
+				casType:  "jiva",
+			},
+		},
+		{
+			"Wrong cas type in all namespace",
+			args{
+				resource: "ResourceType",
+				ns:       "",
+				casType:  "invalid",
+			},
+		},
+		{
+			"Wrong Namespace and all cas types",
+			args{
+				resource: "ResourceType",
+				ns:       "InValidNamespace",
+				casType:  "",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_ = HandleEmptyTableError(tt.args.resource, tt.args.ns, tt.args.casType)
```

Comment:
This is an awesome first test and it likely covers all branches too, but can you compare the error values, i.e. the ones which are returned? That'd be just awesome, for that check the error interface once, you'd find an easy way to compare the expected error from the actual error.

Currently this test doesn't check the return values from the function, i.e. if the code coverage has increased but the real value of unit testing hasn't been put to use to detect if the `system-under-test`(fancy name for the code being tested) 's functionality has changed by a bit and it's callers must be wary of how it's change can affect their functionality.

Reply:
Yes, thanks for the help. Just modified it please have a look.